spacemesh/v2/tx.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+import "google/rpc/status.proto";
+
+package spacemesh.v2;
+
+message Transaction {
+  oneof versioned {
+    TransactionV1 v1 = 1;
+  }
+}
+
+// An immutable Spacemesh transaction.
+// do not include mutable data such as tx state or result.
+message TransactionV1 {
```

Comment:
need to also add recipient and send amount (for Send txs). I'm not 100% sure how to structure this. we should probably add a subfield, something liked ParsedContents, that's `oneof` depending on the type of tx (spawn, self spawn, send, etc.)

Reply:
we can also left that to user

Reply:
Yes, this is an important requirement for consumers. We shouldn't require them to parse a raw TX, the API should provide the parsed TX. If for some reason we think this will be expensive, we can add a flag to the Request object to enable this.